### PR TITLE
Reorder options alphabetically

### PR DESCRIPTION
**Pull request by mhegazy**
order compiler options alphabetically and mark ones that are only used in tsc.ts (e.g. pretty, list files, diagnostics) as internal.

### CODE REVIEW

**Comment by basarat** [comment]
Thanks. Will make keeping [this list updated](https://github.com/alm-tools/alm/blob/ffaebd707ef12e50019db004732ebb5118a766d4/src/server/workers/lang/core/tsconfig.ts#L23-L91) much less painful :) :rose:

File: src/compiler/types.ts [hunk 1]
```diff
@@ -2445,83 +2445,74 @@ namespace ts {
...
-
-        /* @internal */
-        // When options come from a config file, its path is recorded here
-        configFilePath?: string;
-        /* @internal */
-        // Path used to used to compute primary search locations
-        typesRoot?: string;
         types?: string[];
+        /* @internal */ typesRoot?: string;
+        typesSearchPaths?: string[];             // Path used to used to compute primary search locations
```
Comment (sandersn) [inline src/compiler/types.ts]: this comment used to be on typesRoot I think

File: src/compiler/types.ts [hunk 2]
```diff
@@ -2445,83 +2445,74 @@ namespace ts {
...
-        allowSyntheticDefaultImports?: boolean;
-        allowJs?: boolean;
-        noImplicitUseStrict?: boolean;
-        strictNullChecks?: boolean;
-        skipLibCheck?: boolean;
-        listEmittedFiles?: boolean;
-        lib?: string[];
-        /* @internal */ stripInternal?: boolean;
-
-        // Skip checking lib.d.ts to help speed up tests.
```
Comment (sandersn) [inline src/compiler/types.ts]: this comment got lost. Might be OK to drop it though.

**Comment by sandersn** [comment]
:+1: with a couple of comments about comments

File: src/compiler/types.ts [hunk 3]
```diff
@@ -2445,83 +2445,74 @@ namespace ts {
...
     export interface CompilerOptions {
-        allowNonTsExtensions?: boolean;
+        allowJs?: boolean;
+        /*@internal*/ allowNonTsExtensions?: boolean;
+        allowSyntheticDefaultImports?: boolean;
+        allowUnreachableCode?: boolean;
+        allowUnusedLabels?: boolean;
+        baseUrl?: string;
         charset?: string;
+        /* @internal */ configFilePath?: string;           // When options come from a config file, its path is recorded here
```
Comment (yuit) [inline src/compiler/types.ts]: nit: could you reduce space before comment

**Comment by yuit** [comment]
:+1:

File: src/compiler/types.ts (hunk 3 above)
Comment (DanielRosenwasser) [inline src/compiler/types.ts]: Just put the comment above as a JSDoc so you can actually get a tooltip

**Pull request merged**